rs/ledger_suite/icrc1/ledger/src/main.rs

```diff
@@ -225,11 +233,68 @@ fn post_upgrade(args: Option<LedgerArgument>) {
 
     PRE_UPGRADE_INSTRUCTIONS_CONSUMED.with(|n| *n.borrow_mut() = pre_upgrade_instructions_consumed);
 
+    Access::with_ledger_mut(|ledger| {
+        ledger.state = LedgerState::Migrating(LedgerField::Allowances);
+        ledger.clear_arrivals();
+    });
+    migrate_next_part(
+        MAX_INSTRUCTIONS_PER_UPGRADE.saturating_sub(pre_upgrade_instructions_consumed),
+    );
+
     let end = ic_cdk::api::instruction_counter();
     let instructions_consumed = end - start;
     POST_UPGRADE_INSTRUCTIONS_CONSUMED.with(|n| *n.borrow_mut() = instructions_consumed);
 }
 
+fn migrate_next_part(instruction_limit: u64) {
+    STABLE_UPGRADE_MIGRATION_STEPS.with(|n| *n.borrow_mut() += 1);
+    let mut migrated_allowances = 0;
+    let mut migrated_expirations = 0;
+
+    ic_cdk::println!("Migration started.");
```

Comment:
The message sounds like we're at the very beginning of the migration, but this function is actually called repeatedly.

Reply:
I moved this message to the post_upgrade and change this one to reflect that part of the ledger state is being migrated.